unlock-app/src/hooks/useSIWE.tsx

```diff
@@ -79,15 +79,26 @@ export const SIWEProvider = ({ children }: Props) => {
 
       const address = await walletService.signer.getAddress()
       const { data: nonce } = await storage.nonce()
+      const parent = new URL(
+        window.location != window.parent.location
+          ? document.referrer
+          : document.location.href
+      )
+      const resources = []
+      if (parent.host !== window.location.host) {
+        resources.push(window.location.origin)
+      }
       const siwe = new SiweMessage({
-        domain: window.location.hostname,
-        uri: window.location.origin,
+        domain: parent.host,
```

Comment:
Using `host` instead of `hostname` otherwise MM complains about non matching things (they take the port into account)